resolver/type_syntax.cc

```diff
@@ -450,143 +464,147 @@ ParsedSig parseSigWithSelfTypeParams(core::Context ctx, const ast::Send &sigSend
     return sig;
 }
 
-core::TypePtr interpretTCombinator(core::Context ctx, const ast::Send &send, const ParsedSig &sig,
-                                   TypeSyntaxArgs args) {
+TypeSyntax::ResultType interpretTCombinator(core::Context ctx, const ast::Send &send, const ParsedSig &sig,
+                                            TypeSyntaxArgs args) {
     switch (send.fun.rawId()) {
         case core::Names::nilable().rawId():
             if (send.numPosArgs != 1 || send.hasKwArgs()) {
-                return core::Types::untypedUntracked(); // error will be reported in infer.
+                return {.type = core::Types::untypedUntracked()}; // error will be reported in infer.
+            }
+            if (isNilableTProc(ctx, &send)) {
+                auto tproc = getResultTypeAndBindWithSelfTypeParams(ctx, send.args[0], sig, args.withRebind());
+                return {.type = core::Types::any(ctx, tproc.type, core::Types::nilClass()), .rebind = tproc.rebind};
             }
-            return core::Types::any(ctx, getResultTypeWithSelfTypeParams(ctx, send.args[0], sig, args),
-                                    core::Types::nilClass());
+            return {.type = core::Types::any(ctx, getResultTypeWithSelfTypeParams(ctx, send.args[0], sig, args),
+                                             core::Types::nilClass())};
         case core::Names::all().rawId(): {
             if (send.args.empty()) {
                 // Error will be reported in infer
-                return core::Types::untypedUntracked();
+                return {.type = core::Types::untypedUntracked()};
```

Comment:
Could you change these to `TypesSyntax::ResultType{core::Types::untypedUntracked(), nullptr};`? This should be an error as we have `-Werror -Wmissing-field-initializers` in .bazelrc, but it only seems to raise with gcc.

Reply:
yes, though `nullptr` does not work here because the TypeSyntax::ResultType struct includes a full core::SymbolRef object, not a pointer. I believe the default value is `core::SymbolRef()`, so I'll use that to be explicit.